llvm/lib/Telemetry/Telemetry.cpp

```diff
@@ -21,6 +21,8 @@ void TelemetryInfo::serialize(Serializer &serializer) const {
 }
 
 Error Manager::dispatch(TelemetryInfo *Entry) {
+  assert(Config::BuildTimeEnableTelemetry &&
+         "Telemetry should have been enabled");
```

Comment:
Won't this be a problem for unit tests? I see you're enabling them, but I suspect they won't actually succeed with this setting turned off..

Reply:
Is there a way to say "if LLVM_ENABLE_TELEMETRY is not set, then skip the tests)?

Reply:
P.S: How about this? Adding GTEST_SKIP if the static flag is false.

Reply:
done